### PR TITLE
Reduce test duration for google mock CA

### PR DESCRIPTION
**Pull request by ichbinfrog**
Reduce the sleep delay for the mock CA server. The reason for the flakiness is the time it takes until the [first listener accept call](https://github.com/grpc/grpc-go/blob/v1.44.0/server.go#L779) which should not take over 1 second.

Tested on 1000 retries on my local machine.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [x] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: security/pkg/nodeagent/caclient/providers/google/mock/ca_mock.go [hunk 1]
```diff
@@ -62,15 +62,16 @@ func CreateServer(addr string, service *CAService) (*CAServer, error) {
...
 		if err := s.Server.Serve(lis); err != nil {
 			serveErr = err
 		}
 	}()
 
-	// The goroutine starting the server may not be ready, results in flakiness.
-	time.Sleep(1 * time.Second)
+	// The goroutine starting the server may not be ready, results in flakiness
+	// This extra delay ensures that the future grpc dial does not fail in the worst case scenario
+	time.Sleep(100 * time.Millisecond)
```
Comment (howardjohn) [inline security/pkg/nodeagent/caclient/providers/google/mock/ca_mock.go:74]: Any sleep like this at all is a code smell - do we need it at all? It seems to work even with 0s

**Review by howardjohn — APPROVED** [review]

File: security/pkg/nodeagent/caclient/providers/google/mock/ca_mock.go (hunk 1 above)
Comment (ichbinfrog) [reply security/pkg/nodeagent/caclient/providers/google/mock/ca_mock.go:74]: In my tests 0s of delay works perfectly fine. There could be an edge case where the socket isn't registered yet in the routine leading to the test failing but I don't think it warrants this safeguard. 

Removing it completely

**Review by howardjohn — APPROVED** [review]
